chrome/content/zotero/feedSettings.js

```diff
@@ -0,0 +1,172 @@
+
+var Zotero_Feed_Settings = new function() {
+	let urlIsValid = true,
+		data = null,
+		feedReader = null,
+		urlTainted = false;
+	
+	let cleanURL = function(url) {
+		let cleanUrl = Zotero.Utilities.cleanURL(url, true);
+
+		if (cleanUrl) {
+			if (/^https?:\/\/[^\/\s]+\/\S/.test(cleanUrl)) {
+				return cleanUrl;
+			} else {
+				Zotero.debug(uri.scheme + " is not a supported protocol for feeds.");
+			}
+		}
+	};
+	
+	this.init = function() {
+		this.toggleAdvancedOptions(false);
+		
+		data = window.arguments[0];
+		
+		if (data.url) {
+			document.getElementById('feed-url').value = data.url;
+		}
+		
+		if (!data.url) {
+			this.invalidateUrl();
+		} else {
+			// Do not allow to change URL for existing feed
+			document.getElementById('feed-url').readOnly = true;
+		}
+		
+		if (data.title) {
+			document.getElementById('feed-title').value = data.title;
+		}
+		
+		let ttl;
+		if (data.ttl !== undefined) {
+			ttl = Math.floor(data.ttl / 60);
+		} else {
+			ttl = 1;
+		}
+		document.getElementById('feed-ttl').value = ttl;
+		
+		let cleanupAfter = data.cleanupAfter;
+		if (cleanupAfter === undefined) cleanupAfter = 2;
+		document.getElementById('feed-cleanupAfter').value = cleanupAfter;
+		
+		if (data.url && !data.urlIsValid) {
+			this.validateUrl();
+		}
+	};
+	
+	this.invalidateUrl = function() {
+		urlTainted = true;
+		if (feedReader) {
+			feedReader.terminate();
+			feedReader = null;
+		}
+		
+		if (!urlIsValid) return;
+		
+		urlIsValid = false;
+		document.getElementById('feed-title').disabled = true;
+		document.getElementById('feed-ttl').disabled = true;
+		document.getElementById('feed-cleanupAfter').disabled = true;
+		document.documentElement.getButton('accept').disabled = true;
+	};
+	
+	this.validateUrl = Zotero.Promise.coroutine(function* () {
+		if (feedReader) {
+			feedReader.terminate();
+			feedReader = null;
+		}
+		
+		let url = cleanURL(document.getElementById('feed-url').value);
+		urlTainted = false;
+		if (!url) return;
+		
+		try {
+			let fr = feedReader = new Zotero.FeedReader(url);
+			yield fr.process();
+			let feed = fr.feedProperties;
+			// Prevent progress if textbox changes triggered another call to
+			// validateUrl / invalidateUrl (old session)
+			if (feedReader !== fr || urlTainted) return;
+			
+			let title = document.getElementById('feed-title');
+			if (!data.url && feed.title) {
+				title.value = feed.title;
+			}
+			
+			let ttl = document.getElementById('feed-ttl');
+			if (!data.url && feed.ttl) {
+				ttl.value = Math.floor(feed.ttl / 60) || 1;
+			}
+			
+			document.getElementById('feed-url').value = url;
+			
+			urlIsValid = true;
+			title.disabled = false;
+			ttl.disabled = false;
+			document.getElementById('feed-cleanupAfter').disabled = false;
+			document.documentElement.getButton('accept').disabled = false;
+		}
+		catch (e) {
+			Zotero.debug(e);
+		}
+		finally {
+			if (feedReader === fr) feedReader = null;
```

Comment:
This throws `fr is not defined` every time a feed URL is entered. (This would be a case where I personally would just use `var`.)

More generally, there are all sorts of errors being thrown from that dialog, for various inputs:

```
uri is not defined feedSettings.js:45:0

NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIChannel.contentType] FeedProcessor.js:1292:0

NS_ERROR_UNEXPECTED: Component returned failure code: 0x8000ffff (NS_ERROR_UNEXPECTED) [nsISAXXMLReader.onStopRequest] FeedProcessor.js:1298:0

e is undefined bluebird.js:99:0

fr is not defined feedSettings.js:143:0
```